tests/testsuite/build.rs

```diff
@@ -5972,6 +5972,62 @@ fn build_with_relative_cargo_home_path() {
     p.cargo("build").env("CARGO_HOME", "./cargo_home/").run();
 }
 
+#[cargo_test]
+fn skip_mtime_check_in_selected_cargo_home_subdirs() {
+    // Run once without cache
+    let p = project()
+        .at("cargo_home/registry/foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"fn main() { println!("1"); }"#)
+        .build();
+    let project_root = p.root();
+    let cargo_home = project_root
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .to_str()
+        .unwrap();
+    p.cargo("run")
+        .env("CARGO_HOME", &cargo_home)
+        .with_stdout("1")
+        .run();
+    // Run with cache
+    p.change_file("src/main.rs", r#"fn main() { println!("2"); }"#);
+    p.cargo("run")
+        .env("CARGO_HOME", &cargo_home)
+        .with_stdout("1")
+        .run();
```

Comment:
The test looks like. Just some minor details:

* We usually assert `stderr` instead of `stdout`, so that we can avoid `cargo build` and instead use `cargo check`. That makes testsuite run a bit faster.
* When asserting the output for compilation freshness, we usually use `--verbose/-v` to assert `[FRESH]` status unless there is a need of verifying the behavior of final artifacts.
* Changed the `lib.rs` content to some illegal syntax so we can observe whether the build fails or not.
* `env` accepts `AsRef<OsStr>` so `to_str()` is unnecessary.


```suggestion
    let p = project()
        .at("cargo_home/registry/foo")
        .file("Cargo.toml", &basic_lib_manifest("foo"))
        .file("src/lib.rs", "")
        .build();
    let project_root = p.root();
    let cargo_home = project_root
        .parent()
        .unwrap()
        .parent()
        .unwrap();
    p.cargo("check -v")
        .env("CARGO_HOME", &cargo_home)
        .with_stderr(
            "\
[CHECKING] foo v0.5.0 ([CWD])
[RUNNING] `rustc --crate-name foo src/lib.rs [..]
[FINISHED] dev [..]",
        )
        .run();
    // Run with cache
    p.change_file("src/lib.rs", "illegal syntax");
    p.cargo("check -v")
        .env("CARGO_HOME", &cargo_home)
        .with_stderr(
            "\
[FRESH] foo v0.5.0 ([CWD])
[FINISHED] dev [..]",
        )
        .run();
```